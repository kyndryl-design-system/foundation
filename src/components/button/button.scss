/**
 * Copyright Kyndryl, Inc. 2023
 */

@import '../../scss/variables/index';
@import '../../scss/global.scss';

// Base Button Styles
.kd-btn {
  -webkit-transition: border-color 0.3s, color 0.3s, background-color 0.3s;
  transition: border-color 0.3s, color 0.3s, background-color 0.3s;
  font-size: var(--kd-font-size-body-2-md);
  font-weight: var(--kd-font-weight-regular);
  font-family: var(--kd-font-family-secondary);
  line-height: var(--kd-line-height-body-2-md);
  white-space: nowrap;
  border: 1px solid;
  border-radius: 4px;
  text-decoration: none;
  cursor: pointer;
  text-align: left;
  vertical-align: middle;
  align-items: center;
  display: inline-flex;
  flex-direction: row;

  &--small {
    padding: 4px 16px;
    height: 32px;
  }

  &--medium {
    padding: 12px 16px;
    height: 48px;
  }

  &--large {
    padding: 16px 16px;
    height: 56px;
  }
}

// Disabled Button Styles
@mixin kd-btn-disabled {
  background-color: var(--kd-color-background-disabled);
  color: var(--kd-color-text-inversed);
  border: none;
  cursor: not-allowed;
  pointer-events: none;
}

@mixin kd-btn-focus {
  outline: none; // clear default browser outlines
  box-shadow: 0 0 0 2px var(--kd-color-border-focus);
}

// Primary Button Styles (Web & App)
.kd-btn--primary-web,
.kd-btn--primary-app {
  @extend .kd-btn;
  background-color: var(--kd-color-background-primary);
  color: var(--kd-color-text-inversed);
  border: none;

  &:hover {
    background-color: var(--kd-color-background-primary-hover);
  }

  &:active {
    background-color: var(--kd-color-background-primary-pressed);
    border: none;
  }

  &:focus {
    @include kd-btn-focus;
  }

  &:disabled {
    @include kd-btn-disabled;
  }
}

.kd-btn--primary-web {
  background-color: var(--kd-color-background-secondary);

  &:hover {
    background-color: var(--kd-color-background-secondary-hover);
  }

  &:active {
    background-color: var(--kd-color-background-secondary-pressed);
  }
}

// Primary Destructive Button
.kd-btn--primary-destructive {
  @extend .kd-btn;
  background-color: var(--kd-color-background-destructive);
  color: var(--kd-color-text-inversed);
  border: none;

  &:hover {
    background-color: var(--kd-color-background-destructive-hover);
  }

  &:active {
    background-color: var(--kd-color-background-destructive-pressed);
  }

  &:focus {
    @include kd-btn-focus;
  }

  &:disabled {
    @include kd-btn-disabled;
  }
}

// Secondary Button Styles
.kd-btn--secondary {
  @extend .kd-btn;
  background-color: var(--kd-color-transparent);
  color: var(--kd-color-text-primary);
  border-color: var(--kd-color-border-primary);

  &:hover {
    background-color: var(--kd-color-background-secondary-hover);
    color: var(--kd-color-text-inversed);
    border-color: none;
  }

  &:active {
    background-color: var(--kd-color-background-secondary-pressed);
    color: var(--kd-color-text-inversed);
    border-color: transparent;
  }

  &:focus {
    border-color: var(--kd-color-border-focus);
  }

  &:disabled {
    @include kd-btn-disabled;
    background-color: var(--kd-color-transparent);
    color: var( --kd-color-text-disabled);
    border: 1px solid var(--kd-color-border-disabled);
  }
}

// Secondary Destructive Button
.kd-btn--secondary-destructive {
  @extend .kd-btn--secondary;
  color: var(--kd--color-text-destructive);
  border-color: var(--kd-color-border-destructive);

  &:hover, &:active {
    background-color: var(--kd-color-background-destructive-hover);
    color: var(--kd-color-text-inversed);
    border-color: transparent;
  }
}

// Tertiary Button Styles
.kd-btn--tertiary {
  @extend .kd-btn;
  background-color: var(--kd-color-transparent);
  color: var( --kd-color-text-tertiary);
  border: none;

  &:hover, &:active {
    background-color: var(--kd-color-background-accent-stone);
  }

  &:active {
    color: var(--kd-color-text-tertiary-bold);
  }

  &:focus {
    border: 1px solid var(--kd-color-border-focus);
  }

  &:disabled {
    @include kd-btn-disabled;
    background-color: var(--kd-color-transparent);
    color: var( --kd-color-text-disabled);
  }
}

// Tertiary Destructive Button
.kd-btn--tertiary-destructive {
  @extend .kd-btn--tertiary;
  color: var(--kd--color-text-destructive);

  &:hover, &:active, &:focus {
    color: var(--kd--color-text-destructive);
  }
}

// Base styles for button icons
::slotted(kd-icon) {
  svg {
    height: 20px;
    width: 20px;
  }
}

.kd-btn--icon {

  // Styles for buttons with icons on the left side
  &-left {
    // Reverse the flex direction to place the icon to the left
    flex-direction: row-reverse;

    // Styling for the actual icon within the button
    ::slotted(kd-icon) {
      // Add margin to the right to space the icon from the text
      margin-right: 8px;
    }
  }

  // Styles for buttons with icons on the right side
  &-right {
    // Styling for the actual icon within the button
    ::slotted(kd-icon) {
      // Add margin to the left to space the icon from the text
      margin-left: 8px;
    }
  }
}
