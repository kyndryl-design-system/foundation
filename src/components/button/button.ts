import { LitElement, html } from 'lit';
import { customElement, property } from 'lit/decorators.js';
import ButtonScss from './button.scss';

/**
 * Primary UI component for user interaction.
 * @fires on-click - Captures the click event and emits the original event details.
 * @csspart button - Styles to override the button
 */
@customElement('kyn-button')
export class Button extends LitElement {
  static override styles = ButtonScss;

  /** The kind of button to render. */
  @property({ type: String })
  kind = 'primary';

  /** The button size. */
  @property({ type: String })
  size = 'md';

  /** Converts to an &lt;a&gt; tag to create a link. */
  @property({ type: String })
  href = '';

  /** Disables the button. */
  @property({ type: Boolean })
  disabled = false;

  override render() {
    return html`
      ${this.href != ''
        ? html`
            <a
              href=${this.href}
              class="btn btn--${this.kind} btn--${this.size}"
              part="button"
              ?disabled=${this.disabled}
              @click=${(e: Event) => this.handleClick(e)}
            >
              <slot></slot>
            </a>
          `
        : html`
            <button
              class="btn btn--${this.kind} btn--${this.size}"
              part="button"
              ?disabled=${this.disabled}
            >
              <slot></slot>
            </button>
          `}
    `;
  }

  private handleClick(e: Event) {
    const event = new CustomEvent('on-click', {
      detail: { origEvent: e },
    });
    this.dispatchEvent(event);
  }
}

declare global {
  interface HTMLElementTagNameMap {
    'kyn-button': Button;
  }
}
