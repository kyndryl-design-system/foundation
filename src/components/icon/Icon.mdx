import { Meta, Story, Canvas, Controls } from '@storybook/blocks';
import * as IconStories from './Icon.stories.js';
import { stringToReactHtml } from '../../common/helpers/helpers.ts';

import logoutIcon from '../../assets/svg/logout.svg';
import kyndrylIcon from '../../assets/svg/kyndryl-logo.svg';
import bridgeSmallIcon from '../../assets/svg/bridge-logo-small.svg';
import bridgeLargeIcon from '../../assets/svg/bridge-logo-large.svg';

<Meta of={IconStories} />

# Icons

<br />

## Carbon Icons

Most of our icon usage will come from IBM Carbon.

### Framework-specific Icons

It is recommended to use framework-specific versions of the Carbon components.

- [React](https://github.com/carbon-design-system/carbon/tree/main/packages/icons-react)
- [Svelte](https://github.com/carbon-design-system/carbon-icons-svelte)
- [Vue](https://github.com/carbon-design-system/carbon/tree/main/packages/icons-vue)
- [Angular](https://github.com/carbon-design-system/carbon-icons-angular)

### Vanilla Icons

Use the `<kd-icon>` component to make using [vanilla Carbon Icons](https://github.com/carbon-design-system/carbon/tree/main/packages/icons) (bundled with this library) easy. This is more intended for vanilla/Lit usage internally, but should work anywhere.

```js
import '@kyndryl-design-system/shidoka-foundation/components/icon';
import userAvatarIcon from '@carbon/icons/es/user--avatar/24';
```

```html
<kd-icon .icon="${userAvatarIcon}"></kd-icon>
```

<Canvas>
  <Story of={IconStories.Icon} />
</Canvas>

<Controls of={IconStories.Icon} />

### Gallery

[Carbon Icon Library](https://carbondesignsystem.com/guidelines/icons/library/)

<br />

## Kyndryl Icons

Kyndryl icons are custom SVG files that you can import directly from this library.

### Example Usage

It is recommended to use an SVG loader to inject the SVG as inline HTML into your app.

- Example Loaders:
  - [Rollup](https://www.npmjs.com/package/rollup-plugin-svg)
  - [Webpack](https://www.npmjs.com/package/svg-inline-loader)

SVG files can then be imported like this:

```js
import iconName from '@kyndryl-design-system/shidoka-foundation/assets/svg/<filename>.svg';
```

The method for placing them in your app at this point will vary based on the framework/loader.

<br />

### Gallery

<table>
  <thead>
    <tr>
      <th>Icon</th>
      <th>filename</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <span dangerouslySetInnerHTML={stringToReactHtml(logoutIcon)}></span>
      </td>
      <td>logout.svg</td>
    </tr>
    <tr>
      <td>
        <span dangerouslySetInnerHTML={stringToReactHtml(kyndrylIcon)}></span>
      </td>
      <td>kyndryl-logo.svg</td>
    </tr>
    <tr>
      <td>
        <span
          dangerouslySetInnerHTML={stringToReactHtml(bridgeSmallIcon)}
        ></span>
      </td>
      <td>bridge-logo-small.svg</td>
    </tr>
    <tr>
      <td>
        <span
          dangerouslySetInnerHTML={stringToReactHtml(bridgeLargeIcon)}
        ></span>
      </td>
      <td>bridge-logo-large.svg</td>
    </tr>
  </tbody>
</table>
