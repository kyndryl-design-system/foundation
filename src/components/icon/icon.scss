@import '../../common/scss/global.scss';

:host {
  display: inline-block;
}

svg {
  display: block;
}
