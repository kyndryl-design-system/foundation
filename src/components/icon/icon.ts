import { LitElement, html } from 'lit';
import { customElement, property } from 'lit/decorators.js';
import { unsafeHTML } from 'lit-html/directives/unsafe-html.js';
import { getAttributes, toString } from '@carbon/icon-helpers';
import IconScss from './icon.scss';

/**
 * Wrapper for vanilla Carbon icons to simplify usage.
 * @slot unnamed - The default slot for button content.
 */
@customElement('kyn-icon')
export class Icon extends LitElement {
  static override styles = IconScss;

  /** The imported Carbon icon. */
  @property({ type: Object })
  icon: any = {};

  /** Icon fill color. */
  @property({ type: String })
  fill = 'currentColor';

  /** Specify a size in pixels to override the imported Carbon icon's original size. */
  @property({ type: Number })
  sizeOverride = null;

  override render() {
    const attributes = JSON.parse(JSON.stringify(this.icon.attrs));
    attributes.fill = this.fill;

    if (this.sizeOverride) {
      attributes.width = this.sizeOverride;
      attributes.height = this.sizeOverride;
    }

    const iconString = toString({
      ...this.icon,
      attrs: getAttributes(attributes),
    });

    return html` ${unsafeHTML(iconString)} `;
  }
}

declare global {
  interface HTMLElementTagNameMap {
    'kyn-icon': Icon;
  }
}
