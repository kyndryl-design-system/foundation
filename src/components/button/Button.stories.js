/**
 * Copyright Kyndryl, Inc. 2023
 */

// External library imports
import { html } from 'lit';
import { action } from '@storybook/addon-actions';
import chevronRightIcon from '@carbon/icons/es/chevron--right/20';

// Relative component and helper imports
import './button';
import '../icon';
import {
  BUTTON_KINDS,
  BUTTON_SIZES,
  BUTTON_TYPES,
  BUTTON_ICON_POSITION,
} from './defs';
import { createOptionsArray } from '../../common/helpers/helpers';

const createSelectOptions = (defs) => [null, ...createOptionsArray(defs)];

export default {
  title: 'Components/Button',
  component: 'kd-button',
  parameters: {},
  argTypes: {
    type: {
      options: createSelectOptions(BUTTON_TYPES),
      control: { type: 'select', labels: { null: BUTTON_TYPES.BUTTON } },
      table: {
        defaultValue: { summary: BUTTON_TYPES.BUTTON },
      },
    },
    kind: {
      options: createSelectOptions(BUTTON_KINDS),
      control: { type: 'select', labels: { null: BUTTON_KINDS.PRIMARY_APP } },
      table: {
        defaultValue: { summary: BUTTON_KINDS.PRIMARY_APP },
      },
    },
    size: {
      options: createSelectOptions(BUTTON_SIZES),
      control: { type: 'select', labels: { null: BUTTON_SIZES.MEDIUM } },
      table: {
        defaultValue: { summary: BUTTON_SIZES.MEDIUM },
      },
    },
    iconPosition: {
      options: createSelectOptions(BUTTON_ICON_POSITION),
      control: {
        type: 'select',
        labels: { null: BUTTON_ICON_POSITION.CENTER },
      },
      table: {
        defaultValue: { summary: BUTTON_ICON_POSITION.CENTER },
      },
    },
  },
};

const args = {
  unnamed: 'Button Text',
  kind: 'primary-app',
  type: 'button',
  size: 'medium',
  destructive: false,
  disabled: false,
  description: '',
  iconPosition: 'right',
  href: '',
};

export const Button = {
  args,
  render: (args) => {
    return html`
      <kd-button
        kind=${args.kind}
        type=${args.type}
        ?destructive=${args.destructive}
        ?disabled=${args.disabled}
        size=${args.size}
        iconPosition=${args.iconPosition}
        description=${args.description}
        href=${args.href}
        @on-click=${(e) => action(e.type)(e)}
      >
        ${args.unnamed}
      </kd-button>
    `;
  },
};

Button.parameters = {
  design: {
    type: 'figma',
    url: 'https://www.figma.com/file/tTu138pz3jLk0mykJSg3t4/KDS---Foundation-v1.0.0?node-id=4703%3A29021&mode=dev',
  },
};

export const ButtonWithIcon = {
  args,
  render: (args) => {
    return html`
      <kd-button
        kind=${args.kind}
        type=${args.type}
        ?destructive=${args.destructive}
        ?disabled=${args.disabled}
        size=${args.size}
        iconPosition=${args.iconPosition}
        description=${args.description}
        href=${args.href}
        @on-click=${(e) => action(e.type)(e)}
      >
        ${args.unnamed}
        <kd-icon .icon=${chevronRightIcon}></kd-icon>
      </kd-button>
    `;
  },
};

export const IconOnly = {
  args,
  render: (args) => {
    return html`
      <kd-button
        kind=${args.kind}
        type=${args.type}
        ?destructive=${args.destructive}
        ?disabled=${args.disabled}
        size=${args.size}
        iconPosition="center"
        description=${args.description}
        href=${args.href}
        @on-click=${(e) => action(e.type)(e)}
      >
        <kd-icon .icon=${chevronRightIcon}></kd-icon>
      </kd-button>
    `;
  },
};

// Component Variants
// export const Primary_App = Template.bind({});
// Primary_App.args = {
//   kind: BUTTON_TYPES.PRIMARY_APP,
// };

// export const Primary_Web = Template.bind({});
// Primary_Web.args = {
//   type: BUTTON_TYPES.PRIMARY_WEB,
// };

// export const Primary_Destructive = Template.bind({});
// Primary_Destructive.args = {
//   type: BUTTON_TYPES.PRIMARY_APP,
//   destructive: true,
//   size: BUTTON_SIZES.MEDIUM,
// };

// export const Secondary = Template.bind({});
// Secondary.args = {
//   type: BUTTON_TYPES.SECONDARY,
// };

// export const Secondary_Destructive = Template.bind({});
// Secondary_Destructive.args = {
//   type: BUTTON_TYPES.SECONDARY,
//   destructive: true,
// };

// export const Tertiary = Template.bind({});
// Tertiary.args = {
//   type: BUTTON_TYPES.TERTIARY,
// };

// export const Tertiary_Destructive = Template.bind({});
// Tertiary_Destructive.args = {
//   type: BUTTON_TYPES.TERTIARY,
//   destructive: true,
// };

// export const Button_Disabled = Template.bind({});
// Button_Disabled.args = {
//   type: BUTTON_TYPES.PRIMARY_APP,
//   disabled: true,
// };

// export const ButtonWithIcon = {
//   args: {
//     type: BUTTON_TYPES.PRIMARY_APP,
//     iconPosition: 'right',
//   },
//   render: ({ kind, type, size, iconPosition }) => html`
//     <kd-button
//       kind=${kind}
//       type=${type}
//       size=${size}
//       iconPosition=${iconPosition}
//     >
//       Button Text
//       <kd-icon .icon=${chevronRightIcon}></kd-icon>
//     </kd-button>
//   `,
// };

// export const IconOnly = {
//   args: {
//     type: BUTTON_TYPES.PRIMARY_APP,
//     iconPosition: 'center',
//   },
//   render: ({ kind, type, size, iconPosition }) => html`
//     <kd-button
//       kind=${kind}
//       type=${type}
//       size=${size}
//       iconPosition=${iconPosition}
//     >
//       <kd-icon .icon=${chevronRightIcon}></kd-icon>
//     </kd-button>
//   `,
// };
