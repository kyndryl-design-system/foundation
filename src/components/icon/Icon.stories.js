import { html } from 'lit';
import './icon';
import userIcon from '@carbon/icons/es/user--avatar/24';

export default {
  title: 'Components/Icon',
  component: 'kd-icon',
  argTypes: {
    fill: {
      control: { type: 'color' },
    },
    sizeOverride: {
      control: { type: 'number', min: 10, max: 100 },
    },
  },
};

export const Icon = {
  args: {
    icon: userIcon,
    fill: 'currentColor',
    sizeOverride: undefined,
  },
  render: (args) => html`
    <kd-icon
      .icon=${args.icon}
      fill=${args.fill}
      sizeOverride=${args.sizeOverride}
    ></kd-icon>
  `,
};
