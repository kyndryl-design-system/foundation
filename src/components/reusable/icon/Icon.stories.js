import { html } from 'lit';
import './icon';
import userIcon from '@carbon/icons/es/user--avatar/24';

export default {
  title: 'Reusable/Icon',
  component: 'kyn-icon',
  argTypes: {
    fill: {
      control: { type: 'color' },
    },
    sizeOverride: {
      control: { type: 'number', min: 10, max: 100 },
    },
  },
};

export const Icon = {
  args: {
    icon: userIcon,
    fill: 'currentColor',
    sizeOverride: null,
  },
  render: (args) => html`
    <kyn-icon
      .icon=${args.icon}
      fill=${args.fill}
      sizeOverride=${args.sizeOverride}
    ></kyn-icon>
  `,
};
