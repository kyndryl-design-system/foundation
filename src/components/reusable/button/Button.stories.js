import { html } from 'lit';
import '../../reusable/button/button';
import '../../reusable/icon/icon';
import arrowRightIcon from '@carbon/icons/es/arrow--right/16';

export default {
  title: 'Reusable/Button',
  component: 'kyn-button',
  argTypes: {
    kind: {
      control: { type: 'select' },
      options: ['primary', 'secondary'],
    },
    size: {
      control: { type: 'select' },
      options: ['sm', 'md', 'lg'],
    },
  },
};

const args = {
  kind: 'primary',
  size: 'md',
  href: '',
  disabled: false,
};

export const Button = {
  args: args,
  render: (args) => html`
    <kyn-button
      kind=${args.kind}
      href=${args.href}
      size=${args.size}
      ?disabled=${args.disabled}
    >
      Default
    </kyn-button>
  `,
};

export const ButtonWithIcon = {
  args: args,
  render: (args) => html`
    <kyn-button
      kind=${args.kind}
      href=${args.href}
      size=${args.size}
      ?disabled=${args.disabled}
    >
      Button
      <kyn-icon .icon=${arrowRightIcon}></kyn-icon>
    </kyn-button>
  `,
};
