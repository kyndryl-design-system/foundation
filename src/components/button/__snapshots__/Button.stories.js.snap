// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Button Button smoke-test 1`] = `
<kd-button kind="primary-app"
           type="button"
           size="medium"
           iconposition="right"
           description
           href
           name
           value
>
  Button Text
</kd-button>
<button class=" button kd-btn--primary-app kd-btn--medium kd-btn--icon-right "
        type="button"
        aria-label
        title
        name
        value
>
  <span>
    <slot>
    </slot>
    <slot name="icon">
    </slot>
  </span>
</button>
`;

exports[`Components/Button ButtonWithIcon smoke-test 1`] = `
<kd-button kind="primary-app"
           type="button"
           size="medium"
           iconposition="right"
           description
           href
           name
           value
>
  Button Text
  <kd-icon slot="icon">
  </kd-icon>
</kd-button>
<button class=" button kd-btn--primary-app kd-btn--medium kd-btn--icon-right "
        type="button"
        aria-label
        title
        name
        value
>
  <span>
    <slot>
    </slot>
    <slot name="icon">
    </slot>
  </span>
</button>
<svg focusable="false"
     preserveaspectratio="xMidYMid meet"
     xmlns="http://www.w3.org/2000/svg"
     fill="currentColor"
     aria-hidden="true"
     width="20"
     height="20"
     viewbox="0 0 32 32"
>
  <path d="M22 16L12 26 10.6 24.6 19.2 16 10.6 7.4 12 6z">
  </path>
</svg>
`;

exports[`Components/Button IconOnly smoke-test 1`] = `
<kd-button iconposition="center"
           kind="primary-app"
           type="button"
           size="medium"
           description="Button Description"
           href
           name
           value
>
  <kd-icon slot="icon">
  </kd-icon>
</kd-button>
<button class="button kd-btn--primary-app kd-btn--medium kd-btn--icon-center icon-only"
        type="button"
        aria-label="Button Description"
        title="Button Description"
        name
        value
>
  <span>
    <slot>
    </slot>
    <slot name="icon">
    </slot>
  </span>
</button>
<svg focusable="false"
     preserveaspectratio="xMidYMid meet"
     xmlns="http://www.w3.org/2000/svg"
     fill="currentColor"
     aria-hidden="true"
     width="20"
     height="20"
     viewbox="0 0 32 32"
>
  <path d="M22 16L12 26 10.6 24.6 19.2 16 10.6 7.4 12 6z">
  </path>
</svg>
`;
